src/pytorch_lightning/profilers/xla.py

```diff
@@ -50,7 +50,7 @@ def __init__(self, port: int = 9012) -> None:
     def start(self, action_name: str) -> None:
         import torch_xla.debug.profiler as xp
 
-        if action_name in self.RECORD_FUNCTIONS:
+        if action_name.split(".")[-1] in self.RECORD_FUNCTIONS:
```

Comment:
Don't we need to apply the same fix on line 58?

Reply:
Additionally, I suggest adding a comment here:

```suggestion
        # The action name is formatted as '[TYPE]{class name}.{hook name}'
        # Example: [LightningModule]BoringModel.training_step
        if action_name.split(".")[-1] in self.RECORD_FUNCTIONS:
```